### PR TITLE
Design and implement the functionalities for edit page in Process center

### PR DESCRIPTION
**Pull request by isurusuranga**

### CODE REVIEW

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 1]
```diff
@@ -151,6 +152,15 @@ public String createProcess(String processDetails) {
                 appendText(doc, propertiesElement, "bpmnpath", mns, "NA");
                 appendText(doc, propertiesElement, "bpmnid", mns, "NA");
 
+                if(subprocess.length() != 0){
```
Comment (vinok88) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: need a space before curly bracket. add them in other places also.

Comment (isurusuranga) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Fixed in https://github.com/wso2/product-pc/pull/34

Comment (isurusuranga) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Fixed in https://github.com/wso2/product-pc/pull/34

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 2]
```diff
@@ -716,6 +728,92 @@ public String getProcesses() throws ProcessCenterException {
...
+                        predecessor.put("name", predecessorName);
+                        predecessor.put("path", predecessorPath);
+                        predecessor.put("id", predecessorId);
+                        predecessorArray.put(predecessor);
+                    }
+                }
+                resourceString = conObj.toString();
+            }
+        }catch (Exception e) {
+            log.error("Failed to fetch Successor Predecessor and Subprocess information: " + resourcePath);
```
Comment (milindaperera) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: better if log throwable also
log.error("Failed to fetch Successor Predecessor and Subprocess information: " + resourcePath, e);

Comment (isurusuranga) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Fixed in https://github.com/wso2/product-pc/pull/34

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 3]
```diff
@@ -151,6 +152,15 @@ public String createProcess(String processDetails) {
                 appendText(doc, propertiesElement, "bpmnpath", mns, "NA");
                 appendText(doc, propertiesElement, "bpmnid", mns, "NA");
 
+                if(subprocess.length() != 0){
+                    for(int i = 0 ; i < subprocess.length() ; i++){
```
Comment (Kishanthan) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Formatting issue.

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 4]
```diff
@@ -716,6 +728,92 @@ public String getProcesses() throws ProcessCenterException {
...
+
+                        JSONObject subprocess = new JSONObject();
+                        subprocess.put("name", subprocessName);
+                        subprocess.put("path", subprocessPath);
+                        subprocess.put("id", subprocessId);
+                        subprocessArray.put(subprocess);
+                    }
+                }
+
+                if(successorElements.getLength() != 0){
```
Comment (Kishanthan) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Formatting issue.

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 5]
```diff
@@ -716,6 +728,92 @@ public String getProcesses() throws ProcessCenterException {
...
+                    for(int i = 0 ; i < subprocessElements.getLength() ; i++){
```
Comment (Kishanthan) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Formatting issue.

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 4 above)
Comment (isurusuranga) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Already fixed in https://github.com/wso2/product-pc/pull/36 (See under git comment called "Reformat the code")

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 5 above)
Comment (isurusuranga) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Fixed in https://github.com/wso2/product-pc/pull/36 (See under git comment called "Reformat the code")

**Pull request merged**